_posts/2024-06-03-geoserver-tutorials.md

```diff
@@ -0,0 +1,94 @@
+- From the **User, Groups, and Roles** section, select the **Roles** tab. You will find that two roles already exist. They are the administrative roles assigned to the admin account, and they grant access to all GeoServer configurations. Click on the **Edit** link
+- Switch to the **Roles** tab, then click on **Add new role**.
+- Enter `role_reader` as a new role name. We do not need a Parent role. A child role inherits all the grants from the Parent role, making it useful when you want to extend a basic role with more grants. Indeed, we will do this in the next step. 
+- Press the **Save** button and then repeat the previous step to create the `role_editor` role. This time, select "role_reader" as the Parent role.
+- Press the **Save** button and then repeat the previous step to create the `role_admin` role. This time, select "role_editor" as the Parent role.
+- The final step is to associate a role to users or groups. Select the **User, Groups, and Roles** page from the left pane, then select the **Groups** list and click on the "group_readers" group to edit it. Add the "role_reader" role to the group and save it.
+- Now click on the "group_editors" group and associate it with the role_editor role.
+- Finally, associate the "group_admins" group to the "role_admin" role.
+
+By defining roles and associating them to the users, we completed the definition of our organization. Now, we need to explore how data is bound to roles and users.
+
+## Accessing data and services
+GeoServer supports access and control, both at the service level, allowing for the lockdown of service operations to only authenticated users who have been granted a particular role, and on a per-layer basis. The two approaches cannot be mixed. If you lock down a service to a role, you cannot grant the access on a specific layer to the same role. When working with layers, you can define rules that specify what a role can do on any specific layer.
+
+The operations controlled are the view, write, and admin access. When granting read access on a layer, you enable a user to add it on a map; while granting write access you enable the user to update, create, and delete features contained in the layer. The admin access level enables the user to update the layer’s configuration.
+
+## Layer Security
+We want to protect the dataset contained in the `test` workspace from unauthorized access while leaving the remaining layers freely available to all users. In this section, we will associate layers and roles:
+- Navigate to the Data > Security page. The rules list shows the two shipped with the default GeoServer configuration.
+The `*.*.r` rule is associated with the `*` roles. 
+This means that "any user", including the anonymous one, can access "any layer" from "any workspace" configured on GeoServer. The general form of the rule is then as in this code: **workspace.layer.accessMode**.
```

Comment:
`The general form of the rule is then as in this code` -> `The general format of rules is`, or something similar.  `code` is not quite correct here.

Reply:
I fixed it.